utils/clflags.ml

```diff
@@ -167,10 +169,18 @@ let with_runtime = ref true;;         (* -with-runtime *)
 
 let keep_docs = ref false              (* -keep-docs *)
 let keep_locs = ref true               (* -keep-locs *)
-let unsafe_string =
-  if Config.safe_string then ref false
-  else ref (not Config.default_safe_string)
-                                   (* -safe-string / -unsafe-string *)
+
+let get_unsafe_string, set_unsafe_string = (* -safe-string / -unsafe-string *)
+  if Config.safe_string then
+    (fun () -> false),
+    (function
+      | false -> ()
+      | true -> raise (ForbiddenValue
+                         "OCaml has been configured with -force-safe-string: \
+                          unsafe-string is not available"))
```

Comment:
It's slightly unfortunate that `Clflags` (where this option is morally called `-unsafe-string`, given that the module is supposed to be about the command-line flags!) now has the `OCAMLPARAM` name embedded, but this exception can only be raised when parsing that.

The alternative (if it bothers someone else enough) is to keep the `invalid_arg` you had before and do this test in `Compenv` and raise the exception there.